### PR TITLE
[Merged by Bors] - feat(Logic/Embedding/Basic): `setValue_eq_of_ne`

### PR DESCRIPTION
**Pull request by jsm28**
Add a lemma giving the value of `Function.Embedding.setValue` at a point where `setValue` does not change the value.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Logic/Embedding/Basic.lean [hunk 1]
```diff
@@ -180,6 +180,10 @@ theorem setValue_eq_iff {α β} (f : α ↪ β) {a a' : α} {b : β} [∀ a', De
     [∀ a', Decidable (f a' = b)] : setValue f a b a' = b ↔ a' = a :=
   (setValue f a b).injective.eq_iff' <| setValue_eq ..
 
+lemma setValue_eq_of_ne {α β} {f : α ↪ β} {a : α} {b : β} {c : α} [∀ a', Decidable (a' = a)]
+    [∀ a', Decidable (f a' = b)] (hc : c ≠ a) (hb : f c ≠ b) : setValue f a b c = f c := by
+  simp [setValue, hc, hb]
+
```
Comment (b-mehta) [inline Mathlib/Logic/Embedding/Basic.lean:186]: Hmm, it looks like we have a lemma for what happens if c = a, and this adds a lemma for what happens if c != a and f c != b, would you like to also add the lemma for the remaining case where c != a and f c = b?
I suppose this is `setValue f a (f c) c = f a`, and as I write it down, it looks like this one actually holds when c = a as well!

**Review by b-mehta — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/Logic/Embedding/Basic.lean (hunk 1 above)
Comment (jsm28) [reply Mathlib/Logic/Embedding/Basic.lean:186]: I've added this as `setValue_right_apply_eq`, but maybe you have a suggestion for a better name?

Comment (b-mehta) [reply Mathlib/Logic/Embedding/Basic.lean:186]: I can't think of anything better - let's go with this one for now.

Thanks!

**Pull request closed without merging**